src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Options/GraphQLContentOptions.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using System.Linq;
+using OrchardCore.ContentManagement.GraphQL.Settings;
+using OrchardCore.ContentManagement.Metadata.Models;
+
+namespace OrchardCore.ContentManagement.GraphQL.Options
+{
+    public class GraphQLContentOptions
+    {
+        public IEnumerable<GraphQLContentTypeOption> ContentTypeOptions { get; set; }
+            = Enumerable.Empty<GraphQLContentTypeOption>();
+
+        public IEnumerable<GraphQLContentPartOption> PartOptions { get; set; }
+            = Enumerable.Empty<GraphQLContentPartOption>();
+
+        /// <summary>
+        /// Collapsing works at a heirachy
+        ///
+        /// If the Content Type is marked at collapsed, then all parts are collapsed.
+        /// If the Content Type is not marked collapsed, then it falls down to the content type under it.
+        /// If the Content Part at a top level is marked collapsed, then it will trump above.
+        /// </summary>
+        /// <param name="definition"></param>
+        /// <returns></returns>
+        public bool ShouldCollapse(ContentTypePartDefinition definition)
```

Comment:
The other way around. The module that adds the Settings editor will also update the options when the settings are changed, and on startup. The settings will "configure" the option. However using an Option looks weird, why not just a standard service class? Not sure which one is better.

Reply:
Options are fine IMO, I saw settings allowing the user to set it from the UI if not set in code.